File: cpp/test/DynamicKey3_test.cpp
#include "../src/DynamicKey3.h"
#include <gtest/gtest.h>
#include <string>

class DynamicKey3_test : public testing::Test
{
protected:
    virtual void SetUp(){}
    virtual void TearDown(){}

public:
    void test_DynamicKey3();
};

void DynamicKey3_test::test_DynamicKey3(){
    auto staticKey = "970ca35de60c44645bbae84215061b33b";
    auto  signKey = "5cfd2fd1755d40ecb72977518be15d3b";
    auto channelName= "7d72365eb983485397e3e3f9d460bdda";
    auto  unixTs = 1446455472;
    auto  randomInt = 58964981;
    uint32_t uid=2882341273u;
    auto  expiredTs=1446455471;
    std::string result = agora::tools::DynamicKey3::generate(staticKey, signKey, channelName, unixTs, randomInt, uid, expiredTs);
    EXPECT_EQ(result, "00381e1588b09e6e76a0d2e8fd9e5257dbddd9c60a0970ca35de60c44645bbae84215061b33b14464554720383bbf528823412731446455471");
}

TEST_F(DynamicKey3_test, test_DynamicKey3)
{
    test_DynamicKey3();
}

int main(int argc, char* argv[])
{
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: python/test/DynamicKey3Test.py
import sys
import unittest
import os

sys.path.append(os.path.join(os.path.dirname(__file__), '../src'))
from DynamicKey3 import generateDynamicKey3

statickey = "970ca35de60c44645bbae84215061b33b"
signkey = "5cfd2fd1755d40ecb72977518be15d3b"
channelname = "7d72365eb983485397e3e3f9d460bdda"
unixts = 1446455472
uid = 2882341273
randomint = 58964981
expiredts = 1446455471

class DynamicKey3Test(unittest.TestCase):
    def test_generate(self):
        expected = "00381e1588b09e6e76a0d2e8fd9e5257dbddd9c60a0970ca35de60c44645bbae84215061b33b14464554720383bbf528823412731446455471";
        actual = generateDynamicKey3(statickey, signkey, channelname, unixts, randomint, uid, expiredts);
        self.assertEqual(expected, actual)

if __name__ == "__main__":
    unittest.main()
